### PR TITLE
Mini detective investigations update

### PR DESCRIPTION
**Pull request by RemieRichards**
- Replaces hardcoded embedding with new `embed_in()` proc
- Bullets now embed (a dummy object) into a limb, this can only be removed by surgery, it does low damage to compensate for this
- Added `embed_removal_act()` proc which is called when an embedded object is removed, this is only used for dummy bullets at the moment, to prevent them from embedding after being removed
- Dummy bullets contain some rough information about the shot, for example if the bullet was fired less than 5 minutes ago it is described as "warm", there is also a 30% (damage*3 actually) chance the bullet also records how many objects it permutated through (fairly certain this is useless for bullet bullets, but hey)
- Blood now splatters away from an "impact" meaning that if your attacker is facing east, the blood will appear to splatter east, allowing you to roughly determine the direction of murderers
- Item throw speed is now correctly reset after explosions

"Detective investigations" as this is probably mainly useful for the detective, for determining direction of combat and what bullets were used, etc.

### CODE REVIEW

**Comment by Alek2ander** [comment]
Next step: missed bullets embed in walls

**Comment by WJohn** [comment]
What about blood splatters from bullets? Shouldn't that be sent in the opposite direction from the attacker?

**Comment by RemieRichards** [comment]
They are, visually, Mechanically their dir is opposite to them, as if they were coming FROM their own dir.

Also blood splatter from bullets does not produce directional blood as I'm pretty sure bleeding from bullets is from the damage they dealt, not actually the bullet, meaning there is no way to determine direction.

**Comment by SconesC** [comment]
oh christ bullet embeds

**Comment by Miauw62** [comment]
Embed removal surgery should probably be really simple so you can do it a lot of times easily.
Like just scalpel, retract, scalpel, hemostat, cauterize, or even simpler.

**Comment by RemieRichards** [comment]
@Miauw62 it's scalpel bleeder hemo hand, that's really simple.
Not to mention it removes ALL objects in that limb, so if your arm has 5 things, it's still only one surgery,

**Comment by Miauw62** [comment]
Ah, I didnt know so thats ok then.

**Comment by AlexCrimson** [comment]
Probably not gonna see much use. Neat but pointless. You can find out when the bullet was fired by either asking the guy who got shot or using a health scanner to determine the time of death. 

A better feature would be the ability to investigate wounds to see what kind of melee weapon was used to beat someone to death.

**Comment by KorPhaeron** [comment]
The only people who really use bullets are nuke ops, and there isn't much investigation going on then.

If you have the body people are going to just clone it rather than search for bullets in it.
It's a cool concept but it more looks like it'd mess up nuke (nobody on either side has time for surgery) rather than let the detective solve any crimes.

**Comment by Dorsisdwarf** [comment]
Embedded bullets dont appear to actually do anything, they're just for embedding

**Comment by RemieRichards** [comment]
They do minor damage over time (like all embedded objects) and their examine contains information pertaining to their firing.

**Comment by KorPhaeron** [comment]
Also something to consider is the detectives revolver/bartenders shotgun will now send you to surgery instead of being essentially harmless knockdowns. Maybe a force threshhold on embedding bulletd would fix that

**Comment by Iamgoofball** [comment]
@RemieRichards for the blood splatter, check out Hippie's codebase, they have a cool visual  effect of blood flying when you get hit, you could rig it up to work with the blood splatter stuff you  have

**Comment by JJRcop** [comment]
So if I shoot someone and I turn around before the bullet hits them, I can forge false evidence?

**Comment by RemieRichards** [comment]
@JJRcop No, because bullets do not cause blood splatter.
Bullets cause brute damage which causes bleeding which causes blood splatter.
Whereas Physically attacking someone with a melee weapon DOES create a blood splatter on it's own.

tl;dr that's not how it works.

**Comment by JJRcop** [comment]
Okay, maybe it would be a good idea to move embed_in to [/code/game/objects/items.dm](https://github.com/tgstation/-tg-station/blob/112e1d0810f2190c35f7399ef0a5370e9396a1f8/code/game/objects/items.dm) and integrate it into throw_impact

**Comment by RemieRichards** [comment]
... ```throw_impact()``` does use ```embed_in()```, where do you think the hardcode that became ```embed_in()``` came from?

**Comment by JJRcop** [comment]
Oh my bad, I didn't see it... Well now I feel stupid.

**Comment by RemieRichards** [comment]
don't worry, it's more funny that anything, the fact that you became aware of ```embed_in()```s existence, but either misread or didn't read that line of the OP, haha.

**Comment by Alek2ander** [comment]
Rubber bullets/beanbags shouldn't embed. Just as a balance consideration. No need to make the detective's gun more lethal.

**Comment by Dorsisdwarf** [comment]
The detective's gun isn't rubber bullets, though. it's metal. 

Beanbag rounds embedding is dumb though.

**Comment by Supermichael777** [comment]
someday the operating theater will be used. SOME DAY.

**Comment by phil235** [comment]
Merge conflict, plus this really needs a changelog.

**Comment by Razharas** [comment]
I like this but it conflicts
@RemieRichards plz

**Comment by Razharas** [comment]
I guess its dead
Reopen if you will return to it
